packages/datetime/src/dateInput.tsx

```diff
@@ -337,9 +352,18 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
                 this.setState({ isInputFocused: false });
             }
         }
+        this.safeInvokeInputProp("onBlur", e);
     }
 
     private setInputRef = (el: HTMLElement) => {
         this.inputRef = el;
+        const { inputProps = {} } = this.props;
+        Utils.safeInvoke(inputProps.inputRef, el);
+    }
+
+    /** safe wrapper around invoking input props event handler (prop defaults to undefined) */
+    private safeInvokeInputProp(name: keyof HTMLInputProps, e: React.SyntheticEvent<HTMLElement>) {
```

Comment:
hmm it's true this is only meant for the `on*` event handlers, but it's a quick private method and `safeInvoke` will protect us so i'm not sure the extra effort is worth it.